### ISSUE TITLE
Build error

### ISSUE THREAD

**Issue by arBalasquide** [ros-melodic-arch/ros-melodic-image-publisher#1]
Manjaro 5.4.15-2 doing a clean build and I get this error
```Scanning dependencies of target image_publisher_gencfg
Scanning dependencies of target image_publisher_exe
[ 20%] Generating dynamic reconfigure files from cfg/ImagePublisher.cfg: /home/arbalasquide/.cache/yay/ros-melodic-image-publisher/src/build/devel/include/image_publisher/ImagePublisherConfig.h /home/arbalasquide/.cache/yay/ros-melodic-image-publisher/src/build/devel/lib/python3.8/site-packages/image_publisher/cfg/ImagePublisherConfig.py
make[2]: *** No rule to make target '/usr/lib/libpython3.7m.so', needed by 'devel/lib/image_publisher/image_publisher'.  Stop.
make[2]: *** Waiting for unfinished jobs....
[ 40%] Building CXX object CMakeFiles/image_publisher_exe.dir/src/node/image_publisher.cpp.o
Generating reconfiguration files for ImagePublisher in image_publisher
Wrote header file in /home/arbalasquide/.cache/yay/ros-melodic-image-publisher/src/build/devel/include/image_publisher/ImagePublisherConfig.h
[ 40%] Built target image_publisher_gencfg
Scanning dependencies of target image_publisher
make[2]: *** No rule to make target '/usr/lib/libpython3.7m.so', needed by 'devel/lib/libimage_publisher.so'.  Stop.
make[2]: *** Waiting for unfinished jobs....
[ 60%] Building CXX object CMakeFiles/image_publisher.dir/src/nodelet/image_publisher_nodelet.cpp.o
make[1]: *** [CMakeFiles/Makefile2:828: CMakeFiles/image_publisher_exe.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
make[1]: *** [CMakeFiles/Makefile2:126: CMakeFiles/image_publisher.dir/all] Error 2
make: *** [Makefile:130: all] Error 2
==> ERROR: A failure occurred in build().
    Aborting...
Error making: ros-melodic-image-publisher
```

**Comment by jwhendy** [comment]
I can't do this at the moment, but can you try updating the PKGBUILD to match [this one](https://github.com/ros-melodic-arch/ros-melodic-moveit-ros-planning-interface/commit/ea6b51fb966ca3ff0f180681b7615b13492b85df#diff-dfa7cdd9c34798e1af82f172b0822f08)?

Namely:
```
-DPYTHON_INCLUDE_DIR=/usr/include/python3.8 \
-DPYTHON_LIBRARY=/usr/lib/libpython3.8.so \
-DPYTHON_BASENAME=.cpython-38 \
```

There's a list somewhere of all the packages that are actually affected by this. Due to the previous maintainers' build script, almost every `PKGBUILD` includes these links to `python3.x` libs, but almost none of them actually need it. Perhaps this package actually does based on your errors, so hopefully pointing to `libpython3.8.so` does the trick.

**Comment by acxz** [comment]
I did a rebuild and did not get this error. @arBalasquide can you post the full build output?
Maybe try: `sudo rm -rf /usr/lib/python3.7`?

**Comment by arBalasquide** [comment]
@acxz 
```
yay -S ros-melodic-image-publisher
:: Checking for conflicts...
:: Checking for inner conflicts...
[Aur: 1]  ros-melodic-image-publisher-1.12.23-2

  1 ros-melodic-image-publisher              (Installed) (Build Files Exist)
==> Packages to cleanBuild?
==> [N]one [A]ll [Ab]ort [I]nstalled [No]tInstalled or (1 2 3, 1-3, ^4)
==> N
:: PKGBUILD up to date, Skipping (1/1): ros-melodic-image-publisher
  1 ros-melodic-image-publisher              (Installed) (Build Files Exist)
==> Diffs to show?
==> [N]one [A]ll [Ab]ort [I]nstalled [No]tInstalled or (1 2 3, 1-3, ^4)
==> N
:: Parsing SRCINFO (1/1): ros-melodic-image-publisher
==> Making package: ros-melodic-image-publisher 1.12.23-2 (Fri 07 Feb 2020 10:52:27 PM AST)
==> Retrieving sources...
  -> Found ros-melodic-image-publisher-1.12.23.tar.gz
==> Validating source files with sha256sums...
    ros-melodic-image-publisher-1.12.23.tar.gz ... Passed
==> Making package: ros-melodic-image-publisher 1.12.23-2 (Fri 07 Feb 2020 10:52:29 PM AST)
==> Checking runtime dependencies...
==> Checking buildtime dependencies...
==> Retrieving sources...
  -> Found ros-melodic-image-publisher-1.12.23.tar.gz
==> Validating source files with sha256sums...
    ros-melodic-image-publisher-1.12.23.tar.gz ... Passed
==> Removing existing $srcdir/ directory...
==> Extracting sources...
  -> Extracting ros-melodic-image-publisher-1.12.23.tar.gz with bsdtar
==> Sources are ready.
==> Making package: ros-melodic-image-publisher 1.12.23-2 (Fri 07 Feb 2020 10:52:32 PM AST)
==> Checking runtime dependencies...
==> Checking buildtime dependencies...
==> WARNING: Using existing $srcdir/ tree
==> Removing existing $pkgdir/ directory...
==> Starting build()...
-- The C compiler identification is GNU 9.2.0
-- The CXX compiler identification is GNU 9.2.0
-- Check for working C compiler: /usr/sbin/cc
-- Check for working C compiler: /usr/sbin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/sbin/c++
-- Check for working CXX compiler: /usr/sbin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Using CATKIN_DEVEL_PREFIX: /home/arbalasquide/.cache/yay/ros-melodic-image-publisher/src/build/devel
-- Using CMAKE_PREFIX_PATH: /opt/ros/melodic
-- This workspace overlays: /opt/ros/melodic
-- Found PythonInterp: /usr/bin/python3 (found suitable version "3.8.1", minimum required is "2") 
-- Using PYTHON_EXECUTABLE: /usr/bin/python3
-- Using default Python package layout
-- Found PY_em: /usr/lib/python3.8/site-packages/em.py  
-- Using empy: /usr/lib/python3.8/site-packages/em.py
-- Using CATKIN_ENABLE_TESTING: ON
-- Skip enable_testing() when building binary package
-- Using CATKIN_TEST_RESULTS_DIR: /home/arbalasquide/.cache/yay/ros-melodic-image-publisher/src/build/test_results
-- Found gtest: gtests will be built
-- Using Python nosetests: /usr/sbin/nosetests-3.8
-- catkin 0.7.19
-- BUILD_SHARED_LIBS is on
-- Found OpenCV: /usr (found version "4.2.0") found components: core 
-- opencv version 4.2.0
-- Found OpenCV: /opt/opencv3 (found suitable version "3.4.7", minimum required is "3") found components: core imgcodecs videoio 
-- Found PythonInterp: /usr/bin/python3 (found version "3.8.1") 
-- Configuring done
-- Generating done
CMake Warning:
  Manually-specified variables were not used by the project:

    PYTHON_BASENAME
    PYTHON_INCLUDE_DIR
    PYTHON_LIBRARY


-- Build files have been written to: /home/arbalasquide/.cache/yay/ros-melodic-image-publisher/src/build
Scanning dependencies of target image_publisher_gencfg
Scanning dependencies of target image_publisher_exe
make[2]: *** No rule to make target '/usr/lib/libpython3.7m.so', needed by 'devel/lib/image_publisher/image_publisher'.  Stop.
make[2]: *** Waiting for unfinished jobs....
[ 20%] Building CXX object CMakeFiles/image_publisher_exe.dir/src/node/image_publisher.cpp.o
[ 40%] Generating dynamic reconfigure files from cfg/ImagePublisher.cfg: /home/arbalasquide/.cache/yay/ros-melodic-image-publisher/src/build/devel/include/image_publisher/ImagePublisherConfig.h /home/arbalasquide/.cache/yay/ros-melodic-image-publisher/src/build/devel/lib/python3.8/site-packages/image_publisher/cfg/ImagePublisherConfig.py
Generating reconfiguration files for ImagePublisher in image_publisher
Wrote header file in /home/arbalasquide/.cache/yay/ros-melodic-image-publisher/src/build/devel/include/image_publisher/ImagePublisherConfig.h
[ 40%] Built target image_publisher_gencfg
Scanning dependencies of target image_publisher
make[2]: *** No rule to make target '/usr/lib/libpython3.7m.so', needed by 'devel/lib/libimage_publisher.so'.  Stop.
make[2]: *** Waiting for unfinished jobs....
[ 60%] Building CXX object CMakeFiles/image_publisher.dir/src/nodelet/image_publisher_nodelet.cpp.o
make[1]: *** [CMakeFiles/Makefile2:828: CMakeFiles/image_publisher_exe.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
make[1]: *** [CMakeFiles/Makefile2:126: CMakeFiles/image_publisher.dir/all] Error 2
make: *** [Makefile:130: all] Error 2
==> ERROR: A failure occurred in build().
    Aborting...
Error making: ros-melodic-image-publisher
```
This is the full output and I tried deleting /python3.7 but it still yields the same error.

I'll try @jwhendy solution when I have a bit more time.

**Comment by jwhendy** [comment]
This won't necessarily be easy to track down, but I don't think my proposal will work as this package doesn't use those variables:
```
CMake Warning:
  Manually-specified variables were not used by the project:

    PYTHON_BASENAME
    PYTHON_INCLUDE_DIR
    PYTHON_LIBRARY
```

Instead, I think you're being bitten by the need to re-make anything that did depend on those variables. Likely there is a dependency pointing to a python3.7 lib that no longer exists now that arch has updated to python3.8.

It might be worth trying to force a rebuild of all of the packages [listed here](https://github.com/ros-melodic-arch/ros-melodic-desktop-full/issues/8). @acxz I think that's the issue I have in my mind listing all of them... can you confirm?

**Comment by acxz** [comment]
Yeah I think you are right. The culprit is probably ros-melodic-cv-bridge. In any case try rebuilding those packages and if that doesn’t work rebuild all of ROS.

**Comment by acxz** [comment]
Closing this issue, since it has been a week since the last activity.
Feel free to reopen this issue if you are still encountering issues.

**Comment by acxz** [comment]
@arBalasquide putting this here for you in case you haven't gotten this to work:
TLDR: Do the following:
`yay -Rns $(yay -Qsq ros-melodic)`
`rm -rf ~/.cache/yay/ros-melodic-*`
`yay -Syu ros-melodic-image-publisher`

Please let us know if this does not work.

**Comment by arBalasquide** [comment]
@acxz Sorry for no responses for the last week.

I rebuilt ROS and the problem went away, thank you both for the guidance. 
However, how can I avoid these issues in the future?

**Comment by acxz** [comment]
Glad to hear!

Sadly there is no way for you to avoid these problems as of now. We are working on getting prebuilt binaries our ASAP (either through Arch4Edu or the community repo), so that this becomes avoidable, but until then you will have to be patient with us.
